spark_pipeline_framework/transformers/http_data_sender/v2/http_data_sender_processor.py

```diff
@@ -57,7 +63,18 @@ def create_request(
         :param post_as_json_formatted_string: flag to convert the json to json string
         :param json_data: payload for the API
         :param parse_response_as_json flag to parse the response as json
+        :param response_processor: Callable which processes the response
         """
+
+        if response_processor is None:
+
+            def default_behaviour(
+                _: Dict[str, Any], _response: Union[SingleJsonResult, SingleTextResult]
+            ) -> Union[Dict[str, Any], str]:
+                return _response.result
+
+            response_processor = default_behaviour
+
```

Comment:
instead of this, simply have conditional logic -
result=response_processor(json_data, response) if response_processor else json_data